docs/design-docs/cicd-server/comparision_cicd.md

```diff
@@ -0,0 +1,201 @@
+
+### Pipeline as a Code
+__gocd__: possible and looks ok, pipeline code can be in different repository
+
+__jenkins__: possible and looks ok
+
+__jenkins-x__: possible looks ok (Tekton)
+
+__tekton__: pipelines are CRD so can be only as a code
+
+### Build in pods
+
+__gocd__: Elastic agent concepts. Can create many groups (probably on different clusters - not tested yet) and assigned them to proper pipelines
+
+__jenkins__: plugin for building in kubernetes
+
+__jenkins-x__: building in pods in cluster jenkins-x is installed. Possible to install many jenkins-x servers (according to documentation per each team in different namespace). Able to run in multi cluster mode
+
+__tekton__: building in cluster easily. Not possible to build on different server - but I didn't any sence in that use case. Possible to deploy on other kubernetes service.
+
+### Secrets
+__gocd__: Plugins for secrets from: hashicorp vault, kubernetes secrets, file based
+
+__jenkins__: plugins for many options: hashicorp vault, kubernetes secrets, internal secrets, etc
+
+__jenkins-x__: Providers for secrets from: hashicorp vault, kubernetes secrets
+
+__tekton__: Use secrets from kubernetes so everything what is inside kubernetes can be read
+
+### Environment varaibles:
+
+__gocd__: multiple level of variables: environment, pipeline, stage, job
+
+__jenkins__: environment variables can be override
+
+__jenkins-x__: Didn't find any information but expect it will not be worst than in gocd
+
+__tekton__: You can read env variables from any config map so this is kind of overriding.
+
+### Plugins
+
+__gocd__: not big number of plugins (but is this really bad?) but very of them really usefull (LDAP, running in pods, vault, k8s secrets, docker registry, push to S3, slack notification, etc)
+
+__jenkins__: many, many maaaaaaaaaaaany plugins. But if there is too much of them they start making seeeeeeeeeerious issues. Each plugin has different quality and each can breake the server and has its own security issues so we have to be very careful with them.
+
+__jenkins-x__: plugins are called app. There are few of them and this app are helm charts.
+Jenkins-x uses embeded nexus, chartmuseum and monocular services. I don't know if the is option to get rid of them.
+
+__tekton__: tekton itself is kind of plugin for building. You can create whatever you want in different pod and get it.
+
+## Personal conclusion
+__gocd__:
+* This looks like really good CI/CD central server which can be use by many teams.
+* Really mature application. Older release on github from Nov 2014. Accordind to wiki first release in 2007!
+* very intuitive
+* Working really good in kubernetes
+* Good granuality of permission.
+* Good documentation
+* Small amount of help in Internet (compare to jenkins)
+* Small community
+
+GoCD can be easily set up for our organizations. Adding new customers should be big deal. Working with is very intuitive - old school concept of CICD.
```

Comment:
Adding customers should or shouldn't be big deal?

Reply:
fix thanks for that.